### DISCUSSION TITLE
Query regarding post processing

### DISCUSSION THREAD

**Discussion by JRMeyer** [coqui-ai/TTS#76] [Archived]
<USER>>>> alchemi5t<br/> <DATE> [August 19, 2019, 8:15am]<br/> <br/> Hello,<br/> <br/> I have managed to train a model with 13hrs of annotated data. The<br/> alignment is great and the words from the generated test sentences are<br/> easily discernible. The only issue I have is that the generated audio is<br/> not a 100% human-like( a hint of consistent robotic chop-ups in it). I<br/> was wondering if I should fix that with post processing? or could I<br/> handle it with hyper parameter tuning?<br/> <br/> [This is an archived TTS discussion thread from discourse.mozilla.org/t/query-regarding-post-processing]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> carlfm01<br/> <DATE> [August 19, 2019, 4:11pm]<br/> <br/> ![](https://cdn-production.discourse.mozilla-community.org/letter_avatar/alchemi5t/40/5_MOZILLA_1_8504b32b4165840149d2c3dd52eb0686.png){.avatar<br/> <br/> > ( a hint of consistent robotic chop-ups in it)<br/> <br/> With WaveRNN or just TTS?<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> alchemi5t<br/> <DATE> [August 19, 2019, 4:35pm]<br/> <br/> Without wavernn. Just tacotron2 and griffinlim. I am trying out<br/> tacotron2+ erogol's implementation of wavernn right now.<br/> <br/> How much of a difference does wavernn make?<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> carlfm01<br/> <DATE> [August 19, 2019, 4:49pm]<br/> <br/> ![](https://cdn-production.discourse.mozilla-community.org/letter_avatar/alchemi5t/40/5_MOZILLA_1_8504b32b4165840149d2c3dd52eb0686.png){.avatar<br/> <br/> > How much of a difference does wavernn make?<br/> <br/> <https://soundcloud.com/user-565970875/ljspeech-logistic-wavernn> *slash* <br/> WaveRNN using MOL sounds good to me.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> alchemi5t<br/> <DATE> [August 19, 2019, 4:56pm]<br/> <br/> I'll try it out and post the results!<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> erogol<br/> <DATE> [August 20, 2019, 10:50am]<br/> <br/> make sure you disable forward-attention in traning. That might give you<br/> a smoother results. Then you can enable it only at inference.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> alchemi5t<br/> <DATE> [August 20, 2019, 11:58am]<br/> <br/> Sounds good. I'll try that out.<br/> <br/> Regarding your WaveRNN repo, in the ipynb for extracting mel specs, In<br/> generate model outputs( line 37) should that be inside the If condition<br/> for tacotron? Because of it's not, the list mel_specs is empty and you<br/> cannot stack it. And also, the comment for dry_run is the opposite of<br/> what it should be, if I am not wrong. Did I miss something or need these<br/> be corrected?<br/> <br/> Also, great job setting up wavernn! Thank you.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> alchemi5t<br/> <DATE> [August 21, 2019, 4:46am]<br/> <br/> Hey carl!<br/> <br/> Have you tried out WaveRNN? How long does it take to train till 600k<br/> iterations? I am training on one Titan V with 32 workers and it's giving<br/> me around 0.3 steps/sec, which would take over 20 days. is that right?<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> carlfm01<br/> <DATE> [August 21, 2019, 5:03am]<br/> <br/> I'm on it too, no luck yet, attaching an example.<br/> <br/> Still getting familiar with the behavior to know that the things will<br/> work. I'm fine tuning from the mold checkpoint and testing configs.<br/> <br/> > 32 workers<br/> <br/> I think is too much, I use 8 or 6 for a V100.<br/> <br/> Here's the log for 500k from the 400k checkpoint:<br/> <br/> [log-wavernn.zip](https://discourse.mozilla.org/uploads/short-url/cegsXhRD5VXKHQ8hLMwR0MhRSB4.zip)<br/> (70,8 KB)<br/> <br/> [tuxvocoder.zip](https://discourse.mozilla.org/uploads/short-url/uHFzbtKLCwZWCdzM2hrhtsBGAOh.zip)<br/> (84,7 KB)<br/> <br/> Using the defaults params and 30h of clean speech from a single speaker.<br/> I'm not sure if I train it more will succeed, I don't see any<br/> improvement over the steps.<br/> <br/> > around 0.3 steps/sec<br/> <br/> About that, try uninstalling torch and use 0.4.1, sort of remember<br/> getting the same speed<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> alchemi5t<br/> <DATE> [August 21, 2019, 6:42am]<br/> <br/> Thank you for the detailed report. I tried with 6 to 8 workers and that<br/> slowed down my training to *slash* ~ 0.1 steps/sec. From your logs, you were<br/> getting 2.2 steps/sec (Granted that datasets are different, my longest<br/> seq is 420 long, so at best I should've slowed down only *slash* ~3.6(since<br/> default longest seq is 150 long) times, instead of the *slash* ~8 time<br/> retardation). Regardless, I'll keep training it to see how things go.<br/> <br/> ![](../../letter_avatar/carlfm01/40/5_MOZILLA_1_81c48a6156c45fac0d25db91e6a334df.png){.avatar<br/> <br/> > Here's the log for 500k from the 400k checkpoint:<br/> <br/> What language is this? Is the output as expected or is it just<br/> gibberish?<br/> <br/> ![](../../letter_avatar/carlfm01/40/5_MOZILLA_1_81c48a6156c45fac0d25db91e6a334df.png){.avatar<br/> <br/> > 0.4.1<br/> <br/> I'll try this out next.<br/> <br/> Also, if possible, could you share your config file for WaveRNN?<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> carlfm01<br/> <DATE> [August 21, 2019, 6:53am]<br/> <br/> ![](https://cdn-production.discourse.mozilla-community.org/letter_avatar/alchemi5t/40/5_MOZILLA_1_8504b32b4165840149d2c3dd52eb0686.png){.avatar<br/> <br/> > gibberish<br/> <br/> Yes<br/> <br/> ![](https://cdn-production.discourse.mozilla-community.org/letter_avatar/alchemi5t/40/5_MOZILLA_1_8504b32b4165840149d2c3dd52eb0686.png){.avatar<br/> <br/> > What language is this?<br/> <br/> Spanish<br/> <br/> ![](https://cdn-production.discourse.mozilla-community.org/letter_avatar/alchemi5t/40/5_MOZILLA_1_8504b32b4165840149d2c3dd52eb0686.png){.avatar<br/> <br/> > Also, if possible, could you share your config file for WaveRNN?<br/> <br/> Here:<br/> <https://drive.google.com/drive/folders/1wpPn3a0KQc6EYtKL0qOi4NqEmhML71Ve><br/> <br/> Just changed my paths and mel_fmin to 50.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> alchemi5t<br/> <DATE> [August 21, 2019, 8:09am]<br/> <br/> ![](../../letter_avatar/carlfm01/40/5_MOZILLA_1_81c48a6156c45fac0d25db91e6a334df.png){.avatar<br/> <br/> > About that, try uninstalling torch and use 0.4.1<br/> <br/> I got a huge boost from 0.3 steps/sec to *slash* ~0.9 steps/sec. Great catch!<br/> <br/> Just FYI,<br/> <br/> 'batch_size': 64,<br/> 'num_workers': 32,<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> carlfm01<br/> <DATE> [August 21, 2019, 8:12am]<br/> <br/> ![](https://cdn-production.discourse.mozilla-community.org/letter_avatar/alchemi5t/40/5_MOZILLA_1_8504b32b4165840149d2c3dd52eb0686.png){.avatar<br/> <br/> > 'batch_size': 64,<br/> <br/> The default value for the batch_size is 32, 2.2/s steps for me using 32.<br/> fatchord mentioned that no one tried or at least shared results using<br/> values different than 32, keep in mind that it may not work due to the<br/> huge difference from the tested values.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> alchemi5t<br/> <DATE> [August 21, 2019, 8:19am]<br/> <br/> I'll start a parallel run with 32 batch_size and see if that helps. I'll<br/> report results for both.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> carlfm01<br/> <DATE> [August 21, 2019, 8:21am]<br/> <br/> Awesome! Thanks, really helpful for people with limited compute power.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> alchemi5t<br/> <DATE> [August 21, 2019, 4:05pm]<br/> <br/> 30k iterations in and the val loss is around 3.9 but all I hear is loud<br/> static. Did you have similar outputs? Because your log at 400k shows<br/> loss around 4. My inference script might be flawed. I'll share it<br/> tomorrow, please check it out and share insights, if possible .<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> carlfm01<br/> <DATE> [August 21, 2019, 5:44pm]<br/> <br/> ![](https://cdn-production.discourse.mozilla-community.org/letter_avatar/alchemi5t/40/5_MOZILLA_1_8504b32b4165840149d2c3dd52eb0686.png){.avatar<br/> <br/> > 30k iterations<br/> <br/> From checkpoint or from scratch?<br/> <br/> ![](https://cdn-production.discourse.mozilla-community.org/letter_avatar/alchemi5t/40/5_MOZILLA_1_8504b32b4165840149d2c3dd52eb0686.png){.avatar<br/> <br/> > Did you have similar outputs<br/> <br/> Loud static noise, no. Just weird speech, like whisper but fast.<br/> Currently fine tuning taco2 on larger sentences, I'll share results.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> alchemi5t<br/> <DATE> [August 21, 2019, 5:50pm]<br/> <br/> ![](../../letter_avatar/carlfm01/40/5_MOZILLA_1_81c48a6156c45fac0d25db91e6a334df.png){.avatar<br/> <br/> > From checkpoint or from scratch?<br/> <br/> From scratch.<br/> <br/> ![](../../letter_avatar/carlfm01/40/5_MOZILLA_1_81c48a6156c45fac0d25db91e6a334df.png){.avatar<br/> <br/> > taco2 on larger sentences<br/> <br/> Mine is taco2 on fairly large sentences(largest is 420 characters long).<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> carlfm01<br/> <DATE> [August 21, 2019, 5:56pm]<br/> <br/> ![](https://cdn-production.discourse.mozilla-community.org/letter_avatar/alchemi5t/40/5_MOZILLA_1_8504b32b4165840149d2c3dd52eb0686.png){.avatar<br/> <br/> > From scratch.<br/> <br/> Needs more training, there was an issue where mentioned that 200k start<br/> to generate sort of speech. (Of ourse dataset related)<br/> <br/> ![](https://cdn-production.discourse.mozilla-community.org/letter_avatar/alchemi5t/40/5_MOZILLA_1_8504b32b4165840149d2c3dd52eb0686.png){.avatar<br/> <br/> > Mine is taco2 on fairly large sentences(largest is 420 characters<br/> > long).<br/> <br/> How did you fit 420 characters? I can't even fit 290 with 16GB, on the<br/> last steps rush from 6GB to OOM<br/> ![:confused:](<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> alchemi5t<br/> <DATE> [August 23, 2019, 4:13am]<br/> <br/> ![](../../letter_avatar/carlfm01/40/5_MOZILLA_1_81c48a6156c45fac0d25db91e6a334df.png){.avatar<br/> <br/> > Needs more training, there was an issue where mentioned that 200k<br/> > start to generate sort of speech. (Of ourse dataset related)<br/> <br/> That's good news for me.<br/> <br/> ![](../../letter_avatar/carlfm01/40/5_MOZILLA_1_81c48a6156c45fac0d25db91e6a334df.png){.avatar<br/> <br/> > How did you fit 420 characters? I can't even fit 290 with 16GB, on the<br/> > last steps rush from 6GB to OOM<br/> <br/> I trained it with a batch size of 16 and r=2. when i got it to r=1, I<br/> started getting OOM at which point i trained on a batchsize of 8. Like i<br/> said, the output of the TTS was great, just not very humanlike. Words,<br/> pronounciations and intonations were spot on at 150k steps. *slash* [on 13 hrs<br/> of data *slash* ]<br/> <br/> It's consistently static, through different models, steps and settings.<br/> Very good chance the inference script is bad. *slash* <br/> [static.zip](https://discourse.mozilla.org/uploads/short-url/oofjBs4ngzT6Dh1CdjlhVvgFWi5.zip)<br/> (560.9 KB)<br/> <br/> [**[next page<br/> →](https://discourse.mozilla.org/t/query-regarding-post-processing/44070?page=2)**<br/> <br/> [Archived Post]<br/>
